### PR TITLE
Added a warning when the user is on sandbox mode

### LINKED ISSUE #374: Show global notice when sandbox is enabled
## Expected behavior and actual behavior.
The plugin must show a warning global notice when sandbox is enabled. The message can be: "EBANX Gateway - The sandbox mode is enabled. All payments created aren't processed by EBANX."

We can talk to Larissa to help us on this text.

## Specifications like the version of the project, operating system, or hardware.
v1.7.0

### PR DESCRIPTION
**Pull request by andercampanha**
Warning  the user is on sandbox mode
Closes #374

### CODE REVIEW

File: woocommerce-gateway-ebanx/woocommerce-gateway-ebanx.php [hunk 1]
```diff
@@ -121,6 +121,15 @@ public function setup_configs() {
 			$this->is_sandbox_mode = $this->configs->settings['sandbox_mode_enabled'] === 'yes';
 			$this->private_key = $this->is_sandbox_mode ? $this->configs->settings['sandbox_private_key'] : $this->configs->settings['live_private_key'];
 			$this->public_key = $this->is_sandbox_mode ? $this->configs->settings['sandbox_public_key'] : $this->configs->settings['live_public_key'];
+
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/woocommerce-gateway-ebanx.php]: The checking is ok, but I think that we can improve this, for example, we have another better places to put the checking, please, take a look on the method `check_environment`, or we can create/rename some methods. In the future we can have more messages like that. What do you think? 😉

File: woocommerce-gateway-ebanx/woocommerce-gateway-ebanx.php [hunk 2]
```diff
@@ -121,6 +121,15 @@ public function setup_configs() {
...
+			if ($this->is_sandbox_mode) {
+				$warning_message = "You are currently in Sandbox mode, in this mode, none of your transactions will be processed.";
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/woocommerce-gateway-ebanx.php]: When there messages like that, we always use the function `__` and the id `'woocommerce-gateway-ebanx'`. In this case the variable will be: `$warning_message = __('You are currently in Sandbox mode, in this mode, none of your transactions will be processed.', 'woocommerce-gateway-ebanx');`.

Another thing, always use simple quotes :)

File: woocommerce-gateway-ebanx/woocommerce-gateway-ebanx.php [hunk 3]
```diff
@@ -123,6 +125,19 @@ public function setup_configs() {
 			$this->public_key = $this->is_sandbox_mode ? $this->configs->settings['sandbox_public_key'] : $this->configs->settings['live_public_key'];
 		}
 
+		public function check_sandbox_mode() {
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/woocommerce-gateway-ebanx.php]: We need to comment our all methods. Please, comment this one :)

Comment (cezarlz) [inline woocommerce-gateway-ebanx/woocommerce-gateway-ebanx.php]: I think this method can be more generically to support future messages. What do you think?

File: woocommerce-gateway-ebanx/woocommerce-gateway-ebanx.php [hunk 4]
```diff
@@ -124,6 +132,17 @@ public function setup_configs() {
 		}
 
 		/**
+		*
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/woocommerce-gateway-ebanx.php]: I think it's missing something...

File: woocommerce-gateway-ebanx/woocommerce-gateway-ebanx.php [hunk 5]
```diff
@@ -124,6 +132,17 @@ public function setup_configs() {
...
+		*
+		* @return void
+		*/
+		public function checker() {
+
+			WC_EBANX_Checker::check_sandbox_mode($this);
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/woocommerce-gateway-ebanx.php]: Where is the class file?? 😆

**Comment by gpressutto5** [comment]
image

image

**Comment by gpressutto5** [comment]
Can we display the message only if the merchant is on Sandbox Mode AND the sandbox keys are valid?
image

Also, I think it is not a warning, but an info. What do you think @cezarlz ?
One more thing: can we also check if we are accessing it from localhost? I think the best case of performance should be something like this:
`if ( all_keys_are_valid && sandbox_mode && not_localhost )`

**Comment by cezarlz** [comment]
@gpressutto5 I think we can change the text to start with "The Sandbox Mode option is enabled...." and IMO can keep the way it's. I think there aren't differences between info or warning in this case, pick the "more beautiful" for the user.

**Comment by gpressutto5** [comment]
👍  Wait for @cezarlz approval

**Comment by cezarlz** [comment]
Did you resolve the warnings?

**Comment by andercampanha** [comment]
@cezarlz The warnings were shown because I hadn't committed the checker class file. It's working now properly.

**Comment by cezarlz** [comment]
@andercampanha ok, go ahead ;)

**Pull request merged**